src/renderer/utils/createStorage.ts

```diff
@@ -1,73 +1,210 @@
 }
 
-export class StorageHelper<T> {
-  static keyPrefix = "lens_";
+export interface StorageHelperOptions<T> extends StorageConfiguration<T> {
+  autoInit?: boolean; // default: true
+}
 
-  static defaultOptions: IStorageHelperOptions = {
-    addKeyPrefix: true,
-    useSession: false,
-  };
+export interface StorageConfiguration<T> {
+  storage?: StorageAdapter<T>;
+  observable?: CreateObservableOptions;
+}
 
-  constructor(protected key: string, protected defaultValue?: T, protected options?: IStorageHelperOptions) {
-    this.options = Object.assign({}, StorageHelper.defaultOptions, options);
+export interface StorageAdapter<T> {
+  [metadata: string]: any;
+  getItem(key: string): T; // import
+  setItem(key: string, value: T): void; // export
+  removeItem?(key: string): void; // if not provided setItem(key,undefined) will be used
+  onChange?(data: { key: string, value: T, oldValue?: T }): void;
+}
 
-    if (this.options.addKeyPrefix) {
-      this.key = StorageHelper.keyPrefix + key;
+export class StorageHelper<T> {
+  static defaultOptions: StorageHelperOptions<any> = {
+    autoInit: true,
+    get storage() {
+      return jsonFileSyncStorageAdapter;
+    },
+    observable: {
+      deep: true,
+      equals: comparer.shallow,
     }
-  }
+  };
+
+  private data = observable.box<T>();
+  @observable.ref storage: StorageAdapter<T>;
+  @observable initialized = false;
+  whenReady = when(() => this.initialized);
 
-  protected get storage() {
-    if (this.options.useSession) return window.sessionStorage;
+  constructor(readonly key: string, readonly defaultValue?: T, readonly options: StorageHelperOptions<T> = {}) {
+    this.options = { ...StorageHelper.defaultOptions, ...options };
+    this.configure();
+    this.reset();
 
-    return window.localStorage;
+    if (this.options.autoInit) {
+      this.init();
+    }
   }
 
-  get(): T {
-    const strValue = this.storage.getItem(this.key);
+  @action
+  init() {
+    if (this.initialized) return;
 
-    if (strValue != null) {
-      try {
-        return JSON.parse(strValue);
-      } catch (e) {
-        console.error(`Parsing json failed for pair: ${this.key}=${strValue}`);
+    try {
+      const value = this.load();
+      const notEmpty = value != null;
+      const notDefault = !this.isDefaultValue(value);
+
+      if (notEmpty && notDefault) {
+        this.merge(value);
       }
+      this.initialized = true;
+    } catch (error) {
+      console.error(`[init]: ${error}`, this);
     }
+  }
 
-    return this.defaultValue;
+  isDefaultValue(value: T): boolean {
+    return isEqual(value, this.defaultValue);
   }
 
-  set(value: T) {
-    this.storage.setItem(this.key, JSON.stringify(value));
+  @action
+  private configure({ storage, observable }: StorageConfiguration<T> = this.options): this {
+    if (storage) this.storage = storage;
+    if (observable) this.configureObservable(observable);
 
     return this;
   }
```

Comment:
Given the above suggestion and that this leaves this function only as an `if (observable)` I would just move that to the `constructor`.

Furthermore, given the rules of how spreading multiple objects into a new object works. I would unconditionally call `this.configureObservable`.